include/pagmo/problem.hpp

```diff
@@ -564,6 +606,26 @@ inline void check_problem_bounds(const std::pair<vector_double, vector_double> &
                             + " while the upper bound has the smaller value " + std::to_string(ub[i]));
         }
     }
+    // 3 - checks the integer part
+    if (nix) {
+        auto nx = lb.size();
+        auto ncx = nx - nix;
+        if (nix > nx) {
```

Comment:
I woud move the check one line above, before computing ncx. Also, ``const auto``.